data-pipeline/src/trace_exporter.rs

```diff
@@ -213,23 +231,36 @@ impl TraceExporter {
             return Ok(String::from("{}"));
         }
 
+        // todo: do we need to modify the client to allow for &str to avoid allocating a String?
```

Comment:
The client should probably have some sort of `ActionType` enum implemented instead of passing strings like this. Partially for the allocation cost, and partially to reduce the likelihood of typos.